pyads/ads.py

```diff
@@ -1107,44 +1111,92 @@ def notification_decorator(func):
 
             def func_wrapper(notification, data_name):
                 # type: (Any, str) -> None
-                contents = notification.contents
-                data_size = contents.cbSampleSize
-                # Get dynamically sized data array
-                data = (c_ubyte * data_size).from_address(
-                    addressof(contents) + SAdsNotificationHeader.data.offset
+                hNotification, timestamp, value = self.notification_parser(
+                    notification, plc_datatype, timestamp_as_filetime
                 )
+                return func(hNotification, data_name, timestamp, value)
 
-                if plc_datatype == PLCTYPE_STRING:
-                    # read only until null-termination character
-                    value = bytearray(data).split(b"\0", 1)[0].decode("utf-8")
+            return func_wrapper
 
-                elif plc_datatype is not None and issubclass(plc_datatype, Structure):
-                    value = plc_datatype()
-                    fit_size = min(data_size, sizeof(value))
-                    memmove(addressof(value), addressof(data), fit_size)
+        return notification_decorator
 
-                elif plc_datatype is not None and issubclass(plc_datatype, Array):
-                    if data_size == sizeof(plc_datatype):
-                        value = list(plc_datatype.from_buffer_copy(bytes(data)))
-                    else:
-                        # invalid size
-                        value = None
+    def notification_parser(
```

Comment:
I would prefer the function name to be a verb because it is no object but does something. Maybe something like "parse_notification"

Reply:
Makes sense, parse_notification sounds good to me